go.mod

```diff
@@ -1,3 +1,36 @@
 module github.com/luno/lu
 
 go 1.22.3
```

Comment:
Just thinking - if we set 1.22.3 here, it means any users of the package would need that version at a minimum too.

Could we set to 1.21 for now, so it's compatible with more versions? 

Although I guess no one is using it yet, and 1.23 might be out next month, so probably fine 👍

Reply:
i didn't make a conscious choice here, this is what i got from `go mod init ...`
i don't see a reason not to use 1.21 here though.

Reply:
found a reason, some recently added code uses cmp.Or, sticking to 1.22.3

Reply:
All good, since 1.23 is out soon, I don't think this would be an issue anyway 👍